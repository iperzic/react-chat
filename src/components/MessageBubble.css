.MessageBubble, .MessageBubble--own {
    overflow: hidden;
    display: flex;
    padding: 8px 24px;
    border-radius: 5px;
}

.MessageBubble--own {
    justify-content: flex-end;
    align-items: flex-end;
}

.MessageBubble__container, .MessageBubble__container--own {
    box-sizing: border-box;
    max-width: 420px;
    background: white;
    border: 1px solid #d4dade;
    border-radius: 5px;
    order: 1;
    padding: 16px;
}

.MessageBubble__container--own {
    background: #fcf6c5;
}

.MessageBubble__author, .MessageBubble__timestamp, .MessageBubble__timestamp--own {
    color: #b3bcc3;
}

.MessageBubble__timestamp--own {
    margin-right: -8px;
    text-align: right;
}

@media only screen and (max-width : 460px) {
    .MessageBubble__container, .MessageBubble__container--own {
        max-width: 240px;
    }
}
