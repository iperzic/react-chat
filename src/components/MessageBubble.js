import React from 'react';
import PropTypes from 'prop-types';

import './MessageBubble.css';

const MessageBubble = ({
  author, message, timestamp, isOwn,
}) => (
  <div className={isOwn ? 'MessageBubble--own' : 'MessageBubble'}>
    <div className={isOwn ? 'MessageBubble__container--own' : 'MessageBubble__container'}>
      {!isOwn &&
      <div className="MessageBubble__author">
        {author}
      </div>
      }
      <div className="MessageBubble__message">
        {message}
      </div>
      <div className={isOwn ? 'MessageBubble__timestamp--own' : 'MessageBubble__timestamp'}>
        {timestamp}
      </div>
    </div>
  </div>
);

MessageBubble.propTypes = {
  author: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
  timestamp: PropTypes.string.isRequired,
  isOwn: PropTypes.bool.isRequired,
};

export default MessageBubble;
