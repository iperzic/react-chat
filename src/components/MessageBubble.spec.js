import React from 'react';
import { shallow } from 'enzyme';

import MessageBubble from './MessageBubble';

describe('MessageBubble', () => {
  let component;
  let author;
  let message;
  let timestamp;
  let isOwn;

  beforeEach(() => {
    author = 'Bugs Bunny';
    message = 'Whats up doc?';
    timestamp = '12.12.2012';
    isOwn = true;
    component = shallow(<MessageBubble
      author={author}
      message={message}
      timestamp={timestamp}
      isOwn={isOwn}
    />);
  });

  describe('when the message is own', () => {
    it('should display that the message as own', () => {
      expect(component.getElement()).toMatchSnapshot();
    });
  });

  describe('when the message is not own', () => {
    it('should display that the message as not own', () => {
      component.setProps({ isOwn: false });

      expect(component.getElement()).toMatchSnapshot();
    });
  });
});
