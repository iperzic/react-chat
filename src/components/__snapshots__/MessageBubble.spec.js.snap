// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBubble when the message is not own should display that the message as not own 1`] = `
<div
  className="MessageBubble"
>
  <div
    className="MessageBubble__author"
  >
    Bugs Bunny
  </div>
  <div
    className="MessageBubble__message"
  >
    Whats up doc?
  </div>
  <div
    className="MessageBubble__timestamp"
  >
    12.12.2012
  </div>
</div>
`;

exports[`MessageBubble when the message is own should display that the message as own 1`] = `
<div
  className="MessageBubble--own"
>
  <div
    className="MessageBubble__message"
  >
    Whats up doc?
  </div>
  <div
    className="MessageBubble__timestamp--own"
  >
    12.12.2012
  </div>
</div>
`;
